openedx/core/djangoapps/user_api/preferences/tests/test_api.py

```diff
@@ -181,6 +181,34 @@ def test_update_user_preferences(self):
             "new_value"
         )
 
+    def test_update_user_preferences_with_username(self):
```

Comment:
Ah.. Thanks for the detailed explanation.  I interpreted it as such since the implementation of `_get_authorized_user` uses the `allow_staff` parameter.